Added absorbing boundary conditions.

pwessels-uhh opened the pull request:
Added absorbing boundary conditions where particles are removed from the simulation when hitting one boundary of the simulation box by setting charge, velocity and acceleration to zero. This is particularly useful for large simulation boxes (see subsequent pull request).

lucianogsilvestri commented:
Thanks a lot for choosing our code and contributing to it. Absorbing Boundary conditions is great feature that we did not think about it before. I would like to make a suggestion before accepting the PR. It would be nice if we can avoid too many if statement in the time evolution, as such, we would like to have an attribute of the Integrator class, say Integrator.enforce_bc , that links to the chosen boundary conditions. Would you be able to add this feature? As an example see Integrator.update . In the Integrator.setup method Integrator.update is linked to the chosen integrator. In this way, the only if statement is at the initialization process and not during each timestep. We are planning to add reflective boundary conditions as the next step.

pwessels-uhh commented:
Thanks! That is a wise suggestion. I have modified the code accordingly and added an Integrator.enforce_bc attribute that links to "Integrator.absorbing" or "Integrator.periodic" which calls the respective numba functions enforce_abc or enforce_pbc.

lucianogsilvestri commented:
Thanks for making the changes. It looks great.
